### ISSUE TITLE
Automatic update check

### ISSUE THREAD

**Issue by hemmond** [daid/EmptyEpsilon#1366]
@gcask have shown install wizzard for Windows version of Empty Epsilon. I recommend to enhance it further with automatic update check and notification (only in versions where it makes sense). 

For me it makes sense: 
- in installed Windows version
- in .deb package which is manually installed (without repository)

For me it does not make sense:
- in extracted .zip file (when I use zip file, I usually know what version I am running and why this version). 
- in repository-managed packages (I know EE is in Fedora's RPMFusion repository and in Arch Linux package repositories)

This issue is for discussion on how this facility should be implemented so it is not annoying, yet it serves users well. Version mismatch on online events can be a big problem even for other clients, which have correct version.

**Comment by daid** [comment]
It's relatively easy to check if there is a newer version:
https://api.github.com/repos/daid/emptyepsilon/releases/latest

**Comment by gcask** [comment]
Exactly.

I would shove the check behind a preference settings and default it to true.

Package managers are then free to change it to what matches best their respective platform policies.

And if you're bothered by it just disable it from the ini (or the options if you feel like adding a binding there).

**Comment by daid** [comment]
And I think I would preferrer to have this very uninvasive. Like just a box in the corner of the main menu saying [Newer version available: XXXX-XX-XX]

**Comment by hemmond** [comment]
I also prefer uninvasive, but on the other hand, it should be noticed by user. Maybe red notice under version number?

**Comment by gcask** [comment]
I think I prefer @daid 's, something like that maybe?
![image](https://user-images.githubusercontent.com/53709079/112637644-c1645400-8e14-11eb-97d5-29e779e17cae.png)

(don't get too excited, that's just a mock up done in paint.net)

**Comment by daid** [comment]
yeah, something like that. People could have a good reason to run older versions, so making it non intrusive is important IMHO.

**Comment by oznogon** [comment]
I don't like this implementation, which also requires a change to SeriousProton to send a `User-Agent` header in the request (otherwise GitHub forbids it): https://gist.github.com/oznogon/c7dce727388abead734c7b0284e0a4b6

But it shows that we can poll the GitHub API with `sp::io::http::Request`, parse with `sp::json::parse`, compare with VERSION_NUMBER, and flag it in the log and main menu.

![image](https://user-images.githubusercontent.com/19192104/212770230-fdf82ad8-589e-4d87-841b-2dc6561d17d8.png)
